app/src/ui/repositories-list/index.tsx

```diff
@@ -65,7 +67,14 @@ export class RepositoriesList extends React.Component<IRepositoriesListProps, vo
   }
 
   public render() {
+    if (this.props.repositories.length > 0) {
+      this.hadRepos = true
+    }
+
     if (this.props.repositories.length < 1) {
+      if (this.hadRepos) {
+        this.props.dispatcher.closeFoldout(FoldoutType.Repository)
```

Comment:
`render` methods should, as a rule, avoid doing any sort of side effects. They should be pure functions of their state and props.

Instead of closing the foldout, we should change the logic that opens it in the first place. See https://github.com/desktop/desktop/blob/49a671fe9836ccd44201c97d34d49708c2f9c4a3/app/src/lib/dispatcher/dispatcher.ts#L174. Instead of always showing the foldout, we should only show it if there are still repositories after removing the ones the user asked us to.